packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -130,13 +130,30 @@ abstract class StateMachineBase extends Resource implements IStateMachine {
   public static fromStateMachineArn(scope: Construct, id: string, stateMachineArn: string): IStateMachine {
     class Import extends StateMachineBase {
       public readonly stateMachineArn = stateMachineArn;
+      public readonly executionArn = Arn.format({
+        resource: 'execution',
+        service: 'states',
+        account: this.stack.account,
+        partition: this.stack.partition,
+        region: this.stack.region,
+        resourceName: Arn.parse(stateMachineArn, ':').resourceName,
+        sep: ':',
+      }, this.stack);
```

Comment:
does a state machine need to expose the `executionArn` as a public property? Is it only being used internally? the executionArn seems to be a function of the stateMachineName. What's the scenario where a user would need an execution ARN?

I don't think we should make this a property.

aside: you likely don't want to use `Arn.format` directly and would use `Stack.of(this).formatArn({...})`... and you'll get to drop the account, partition, region specification as those would be derived from the stack

can the arn pattern be constructed and used internally instead?
`arn:<partition>:states:<region>:<account-id>:execution:<stateMachineName>:*`

Reply:
executionArn is now private.
Arn.format() is swapped to Stack.of(this).formatArn()